src/os/bluestore/BlueStore.cc

```diff
@@ -17532,7 +17532,9 @@ int BlueStore::_do_write_v2(
   if (bl.length() != length) {
     bl.splice(length, bl.length() - length);
   }
-  o->extent_map.fault_range(db, offset, length);
+  uint64_t start = p2align(offset, min_alloc_size);
+  uint64_t end = round_up_to(offset + length, min_alloc_size);
```

Comment:
round_up_to is costly, it % and *, here we can use p2roundup.

Reply:
updated.